ast/marshal/json.go

```diff
@@ -0,0 +1,31 @@
+package marshal
```

Comment:
This has to be a new package since it's used in ast and ast/location and we need to avoid the cycle